lib/ClangImporter/ImportDecl.cpp

```diff
@@ -3639,6 +3639,29 @@ namespace {
             return nullptr;
           }
         }
+
+        for (const auto method : methods) {
```

Comment:
I'm not sure this is the correct place for this logic. Why not do this when importing the method itself?

Reply:
If we were able to combine this with the logic for importing methods, then we could also get rid of `cxxSubscripts`, I think.

Reply:
Basically having a single `operator[]` method isn't always enough to generate a subscript. I don't think we can easily get rid of `cxxSubscripts`, here's why:

Imagine we're importing a const `operator[]` with a parameter type that wasn't used in any of the already imported  `operator[]` methods. This method always corresponds to a subscript getter, and a subscript with this getter is always valid, so we could just add a getter-only subscript and move on, and later if we import a non-const `operator[]` with the same parameter type, we'll add a setter to the existing subscript.

But if we get a non-const `operator[]` with an unseen before parameter type, there's no obvious way to proceed. We can't add a subscript right away, because setter-only subscripts aren't valid in Swift. We could create a new subscript with both accessors using the non-const `operator[]`, and then generate a new getter from a const `operator[]` if we see one, but this would be suboptimal for structs that have both const & non-const `operator[]` – we'd be generating getters only to throw them away later.

So storing these methods in `cxxSubscripts` and generating subscript decls after all the members have been imported seems to be the easiest & most efficient solution. WDYT?

Reply:
Thanks for explaining. This logic makes sense now. 

One suggestion about this loop, though. And this is a super small thing: would it make more sense to iterate over `cxxSubscripts` rather than all the methods in the record? I think we can do something along the lines of:
```
for (auto &keyValue : Impl.cxxSubscripts) {
  if (keyValue.first.first != result)
    continue;

  auto getterAndSetter = keyValue.second;
  auto subscript = makeSubscript(getterAndSetter.first,
                                 getterAndSetter.second);
  result->addMember(subscript);
  Impl.cxxSubscripts.erase(keyValue.first);
}
```
(Maybe with better variable names.)

Reply:
That looks reasonable to me, fixed it.

Reply:
I've removed the `Impl.cxxSubscripts.erase` statement because is is no longer needed (one `cxxSubscripts` entry corresponds to no more than one subscript by definition), and to keep it from invalidating the `cxxSubscripts` iterators (since we're now iterating over `cxxSubscripts` itself).